### PR TITLE
Croissant 1.0 spec & RAI vocabulary turtle definition

### PR DESCRIPTION
**Pull request by benjelloun**

### CODE REVIEW

File: docs/croissant_rai.ttl [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+rdfs:label "dataCollection" ;
+rdfs:comment "Description of the data collection process." ;
+schema:domainIncludes schema:Dataset ;
+schema:rangeIncludes schema:Text .
+
+rai:dataCollectionType a rdf:Property ;
+rdfs:label "dataCollectionType" ;
+rdfs:comment "Define the data collection type. Recommended values: Surveys, Secondary Data analysis, Physical data collection, Direct measurement, Document analysis, Manual Human Curator, Software Collection, Experiments, Web Scraping, Web API, Focus groups, Self-reporting, Customer feedback data, User-generated content data, Passive Data Collection, Others." ;
+schema:domainIncludes schema:Dataset ;
+schema:rangeIncludes schema:Text, schema:ItemList .
```
Comment (benjelloun) [inline docs/croissant_rai.ttl:19]: How should ItemList be used for values of this property? Alternatively, values can just be a list of strings, e.g.:

```
"dataCollectionType": ["Surveys", "Secondary Data analysis"]
```

Following schema.org, this is allowed if you just specify the range to be Text.

This comment applies to all the ItemList instances in this turtle file.

File: docs/croissant_rai.ttl [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+rdfs:label "dataCollectionRawData" ;
+rdfs:comment "Description of the raw data i.e. source of the data." ;
+schema:domainIncludes schema:Dataset ;
+schema:rangeIncludes schema:Text, schema:ItemList .
+
+rai:dataCollectionTimeframe a rdf:Property ;
+rdfs:label "dataCollectionTimeframe" ;
+rdfs:comment "Timeframe in terms of start and end date of the collection process." ;
+schema:domainIncludes schema:Dataset ;
+schema:rangeIncludes schema:DateTime .
```
Comment (benjelloun) [inline docs/croissant_rai.ttl:37]: This should specify how the time interval should be represented, e.g., following https://en.wikipedia.org/wiki/ISO_8601#Time_intervals. 

https://schema.org/temporalCoverage might be a good example to follow.

File: docs/croissant_rai.ttl [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+rai:dataAnnotationAnalysis a rdf:Property ;
+rdfs:label "dataAnnotationAnalysis" ;
+rdfs:comment "Considerations related to the process of converting the “raw” annotations into the labels that are ultimately packaged in a dataset - Uncertainty or disagreement between annotations on each instance as a signal in the dataset, analysis of systematic disagreements between annotators of different socio demographic group,  how the final dataset annotations will relate to individual annotator responses." ;
+schema:domainIncludes schema:Dataset ;
+schema:rangeIncludes schema:Text .
+
+rai:annotationsPerItem a rdf:Property ;
+rdfs:label "annotationsPerItem" ;
+rdfs:comment "Number of human labels per dataset item." ;
```
Comment (benjelloun) [inline docs/croissant_rai.ttl:89]: Consider adding: :", if applicable"

File: docs/croissant_rai.ttl (hunk 1 above)
Comment (JoanGi) [reply docs/croissant_rai.ttl:19]: We do not have a complete taxonomy of collection types. We discussed giving some recommended options but allowing users freedom (also following the schema common criteria). This freedom will let us look at users' behavior in the future and maybe recommend a more closed item list.  

So, range could be schema:Text

Comment (omshinde) [reply docs/croissant_rai.ttl:19]: I copied the `range` from `ExpectedType column` of the table "RAI property information" present in the RAI spec.

File: docs/croissant_rai.ttl (hunk 2 above)
Comment (omshinde) [reply docs/croissant_rai.ttl:37]: Thanks. 
 We can add `schema:supersededBy schema:temporalCoverage ;` for defining time frame

Comment (mubasharaak) [reply docs/croissant_rai.ttl:37]: thanks for the comments, is it best to specify using rdfs:comment that the time interval should be indicated in ISO 8601 interval format?

File: docs/croissant_rai.ttl (hunk 1 above)
Comment (benjelloun) [reply docs/croissant_rai.ttl]: I added suggestions to update the spec.

File: docs/croissant_rai.ttl (hunk 3 above)
Comment (benjelloun) [reply docs/croissant_rai.ttl]: Applied the fixes for previous issues that Mubashara created. PTAL.

File: docs/croissant_rai.ttl [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+
+rai:annotatorDemographics a rdf:Property ;
+rdfs:label "annotatorDemographics" ;
+rdfs:comment "List of demographics specifications about the annotators, if applicable." ;
+schema:domainIncludes schema:Dataset ;
+schema:rangeIncludes schema:Text .
+
+rai:machineAnnotationTools a rdf:Property ;
+rdfs:label "machineAnnotationTools" ;
+rdfs:comment "List of software used for data annotation ( e.g. concept extraction, NER, and additional characteristics of the tools used for annotation to allow for replication or extension)." ;
```
Comment (marcenacp) [inline docs/croissant_rai.ttl:101]: `( e.g.` -> `(e.g.`

File: docs/croissant_rai.ttl [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+
+rai:dataAnnotationPlatform a rdf:Property ;
+rdfs:label "dataAnnotationPlatform" ;
+rdfs:comment "Platform, tool, or library used to collect annotations by human annotators, if applicable." ;
+schema:domainIncludes schema:Dataset ;
+schema:rangeIncludes schema:Text .
+
+rai:dataAnnotationAnalysis a rdf:Property ;
+rdfs:label "dataAnnotationAnalysis" ;
+rdfs:comment "Considerations related to the process of converting the “raw” annotations into the labels that are ultimately packaged in a dataset - Uncertainty or disagreement between annotations on each instance as a signal in the dataset, analysis of systematic disagreements between annotators of different socio demographic group,  how the final dataset annotations will relate to individual annotator responses." ;
```
Comment (marcenacp) [inline docs/croissant_rai.ttl:83]: Not sure how `“` (vs `"`) will be rendered if eg you generate HTML documentation from this turtle

**Review by marcenacp — APPROVED** [review]
Thanks!

**Review by marcenacp — COMMENTED** [review]
Can you use https://docs.google.com/drawings/d/1PeipmbFgqWVEYYlxzoYcZJL45WPjh3kk_o4C5urAqiM/edit for `docs/images/cross-product.png`? Thanks!

File: docs/croissant-spec.md [hunk 6]
```diff
@@ -1,467 +1,2110 @@
...
+
+# Dataset-level Information
+
+## schema.org/Dataset
+
+Croissant builds on the [schema.org/Dataset](http://schema.org/Dataset) vocabulary, which is widely adopted by datasets on the web. An introduction to describing datasets with this vocabulary can be found [here](https://developers.google.com/search/docs/appearance/structured-data/dataset).
+
+[Schema.org](http://Schema.org) properties are known to be very flexible in terms of the types of values they accept. We list below the main properties of the vocabulary and their expected type.To facilitate more consistent use of these properties we provide additional constraints on their usage in the context of Croissant datasets. We also specify cardinalities to clarify if a property can take one or many values.
+
+We organize [schema.org](http://schema.org) properties in three categories: Required, recommended and other properties. The properties starting with the symbol @ are defined in JSON-LD, which is our RDF syntax of choice for Croissant.
```
Comment (pierrot0) [inline docs/croissant-spec.md:331]: I would put the "@" within bacquotes: ``@``

File: docs/croissant-spec.md [hunk 7]
```diff
@@ -1,467 +1,2110 @@
...
+   <th>Comments</th>
+  </tr>
+  <tr>
+   <td>@context
+   </td>
+   <td><a href="http://schema.org/URL">URL</a>
+   </td>
+   <td>ONE
+   </td>
+   <td>A set of JSON-LD context definitions that make the rest of the Croissant description less verbose. See the recommended JSON-LD context in Appendix 1.
```
Comment (pierrot0) [inline docs/croissant-spec.md:354]: [Appendix 1](#appendix-1-json-ld-context)

File: docs/croissant-spec.md [hunk 8]
```diff
@@ -1,467 +1,2110 @@
...
+}
+```
+
+# RecordSets
+
+While `FileObject` and `FileSet` describe the resources contained in a dataset, they do not tell us anything about the way the content within the resources is organized. This is the role of `RecordSet`.
+
+A key challenge is that ML data comes in many different formats, including unstructured formats such as text, audio and video, and structured ones such as CSV and JSON. All these formats, no matter their level of machine-readable structuredness, need to be loaded into a common representation for ML purposes, and sometimes combined despite their heterogeneity.
+
+`RecordSet` provides a common structure description that can be used across different modalities, in terms of records_ _that may contain multiple fields_._ Unstructured content, like text and images, is represented as single-field records. Tabular data yields one record per row in the table, with fields for each column. Tree-structured data can be described with nested and repeated fields.
```
Comment (pierrot0) [inline docs/croissant-spec.md:965]: The rendering of that is a bit strange.
http://screen/48d7i2TWYeeETVn

File: docs/croissant-spec.md [hunk 9]
```diff
@@ -1,467 +1,2110 @@
...
-**domain**: [Field](#field)
 
-### data
+## Categorical Data
 
-In a `RecordSet` object, this property defines the RecordSet data inline. This
-inline mechanism is only expected to be used with small `RecordSet` instances,
-for example the ones defining enumerated values. In such a case, data is a JSON
-list, and each object within that list must define the RecordSet fields.
+In machine learning applications, it's often useful to know that some of the data is categorical in nature, and has a finite set of values that can be used, say, for classification. Croissant represents that information by using the [sc:Enumeration](schema.org/Enumeration) class from [schema.org](Schema.org), as a `dataType` on `RecordSet`s that hold categorical data.
```
Comment (pierrot0) [inline docs/croissant-spec.md:1838]: http://schema.org/Enumeration

Comment (pierrot0) [inline docs/croissant-spec.md:1838]: `[schema.org](Schema.org) => [Schema.org](http://schema.org)`

**Review by pierrot0 — APPROVED** [review]

**Pull request merged**